src/invariant/ConservationOfLumens.cpp

```diff
@@ -36,6 +42,34 @@ calculateDeltaBalance(LedgerEntry const* current, LedgerEntry const* previous)
         return ((current ? current->data.claimableBalance().amount : 0) -
                 (previous ? previous->data.claimableBalance().amount : 0));
     }
+    case LIQUIDITY_POOL:
+    {
+        auto const* currentBody =
+            current ? &current->data.liquidityPool().body.constantProduct()
+                    : nullptr;
+        auto const* previousBody =
+            previous ? &previous->data.liquidityPool().body.constantProduct()
+                     : nullptr;
+
+        auto const& assetA =
```

Comment:
not sure if we care, but `LedgerEntryIsValid::checkIsValid` does not enforce that lp's parameters are immutable and this code relies on that

Reply:
Good idea. Made some general improvements to that invariant.